### DISCUSSION TITLE
Letter formatting in csv

### DISCUSSION THREAD

**Discussion by Phoro92** [kanboard/kanboard#5180] [General]
Hello,

I have a problem with formatting Polish characters when exporting to csv. Everything is ok on the board, the problem occurs only when I open the csv file in Excel. What is interesting, when I open csv file using notepad or notepad++ Polish characters are displayed correctly. Does anyone know what the problem is or how to fix it?

How is in board view:
image
![image](https://user-images.githubusercontent.com/68482509/197061151-d982196e-f69e-4076-9af8-4b4b32637018.png)

What i get in excel:
Polish letters: Ä…, Ä‡, Ä™, Ĺ‚, Ĺ„, Ăł, Ĺ›, Ĺş, ĹĽ

On the forum I got help in the form of this link where the second method works.
https://www.accompa.com/kb/answer.html?answer_id=264

In addition, if I download a csv and edit it in notepad change the encoding to ANSI and save, after reopening the file Polish letters are displayed correctly. I use Microsoft office 365
It seemed to me that the file must be encoded in utf-8 to have non-english letters, and changing the encoding in notepad to ANSI helped, strange.

**Comment by fguillot** [comment]
Kanboard already exports the CSV files in UTF-8. You have to explicitly tell Excel to open the CSV file in UTF-8 instead of the default Windows encoding.

**Comment by Phoro92** [comment]
i found a solution, in file app\Core\csv.php, i think bom was missing i added in write section line 110, like this and now is working:

public static function output(array $rows)
{
        $csv = new static;
        $csv->write('php://output', $rows);
}

To this:

public static function output(array $rows)
{
    $csv = new static;
    header('Content-Type: text/csv; charset=utf-8');
    header('Content-Disposition: attachment; filename="'.date("Y-m-d").'_export.csv"');
    echo "\xEF\xBB\xBF"; // add BOM
    $csv->write('php://output', $rows);
}

**Comment by jnahmias** [comment]
@korlikiangel - you can make UTF-8 the default encoding in Excel. Instructions are here: https://superuser.com/a/1179248/318203
